File: 0443-string-compression/0443-string-compression.cpp
class Solution {
public:
    int compress(vector<char>& chars) {
        int n = chars.size();
        int length = 0;
        
        int i = 0;
        vector<int> newChars;
        while(i < n)
        {
            char curr = chars[i];
            newChars.push_back(curr);
            i++, length++;
            int freq = 1;
            while(i < n && chars[i] == curr)
                i++, freq++;
            if(freq > 1)
            {
                string freq_s = to_string(freq);
                for(int j = 0; j < freq_s.length(); j++)
                    newChars.push_back(freq_s[j]);
                length += freq_s.length();
            }
        }
        while(chars.size())
            chars.pop_back();
        chars.resize(newChars.size());
        // chars = newChars;
        for(int i = 0; i < newChars.size(); i++)
            chars[i] = newChars[i];
        return length;
    }
};

File: 0443-string-compression/0443-string-compression.java
class Solution {
    public int compress(char[] chars) {
        int n = chars.length;
        
        int i = 0;
        String ans = "";
        while(i < n)
        {
            int idx = i;
            int count = 0;
            while(i < n && chars[i] == chars[idx]) {
                i++;
                count++;
            }
            if(count == 1) {
                ans += chars[idx];
            }
            else {
                ans += chars[idx];
                ans += Integer.toString(count);
            }
        }
        
        for(i = 0; i < ans.length(); i++) {
            chars[i] = ans.charAt(i);
        }
        
        return ans.length();
    }
}